ui/src/app/views/workspace/component.ts

```diff
@@ -217,8 +218,20 @@ export class WorkspaceComponent implements OnInit {
     this.router.navigate(['share'], {relativeTo : this.route});
   }
 
-  hasPermission(): boolean {
+  delete(): void {
+    this.deleting = true;
+    this.workspacesService.deleteWorkspace(
+        this.workspace.namespace, this.workspace.id).subscribe(() => {
+          this.router.navigate(['/']);
+        });
+  }
+
+  hasWritePermission(): boolean {
```

Comment:
I still think this is the ideal type of place to use [get](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get).  If you disagree though I'll stop harping on it ;)